### PR TITLE
Pkg controller test requests 5

### PR DESCRIPTION
**Pull request by bgeuken**

### CODE REVIEW

File: src/api/spec/controllers/webui/request_controller_spec.rb [hunk 1]
```diff
@@ -76,4 +76,44 @@
...
     end
   end
+
+  describe "POST #change_devel_request" do
+    context "with valid parameters" do
+      before do
+        login(submitter)
+        post :change_devel_request, project: target_project.name, package: target_package.name,
+          devel_project: source_project.name, devel_package: source_package.name, description: "change it!"
+        @bs_request = BsRequest.where(description: "change it!", creator: submitter.login, state: "new").first
```
Comment (hennevogel) [inline src/api/spec/controllers/webui/request_controller_spec.rb]: No factory?

File: src/api/spec/controllers/webui/request_controller_spec.rb [hunk 2]
```diff
@@ -76,4 +76,44 @@
...
+        expect(request_action).to exist
+      end
+    end
+
+    context "with invalid devel_package parameter" do
+      before do
+        login(submitter)
+        post :change_devel_request, project: target_project.name, package: target_package.name,
+          devel_project: source_project.name, devel_package: "non-existant", description: "change it!"
+        @bs_request = BsRequest.where(description: "change it!", creator: submitter.login, state: "new").first
```
Comment (hennevogel) [inline src/api/spec/controllers/webui/request_controller_spec.rb]: Same here, don't we have a factory for this?

File: src/api/spec/controllers/webui/request_controller_spec.rb (hunk 1 above)
Comment (bgeuken) [inline src/api/spec/controllers/webui/request_controller_spec.rb]: That's a query;-) Just checking that the BsRequest got created and is correct.

Comment (hennevogel) [inline src/api/spec/controllers/webui/request_controller_spec.rb]: right. stupid me...

File: src/api/test/functional/webui/package_controller_test.rb [hunk 3]
```diff
@@ -284,6 +284,8 @@ def test_delete_request
     click_button 'Revoke request'
   end
 
+  # spec/features/webui/packages_spec.rb
+  # spec/controllers/webui/request_controller_spec.rb
   def test_change_devel_request
```
Comment (hennevogel) [inline src/api/test/functional/webui/package_controller_test.rb]: Can you put at least one beside the `def ...` so we can filter which tests are migrated please.

Comment (bgeuken) [inline src/api/test/functional/webui/package_controller_test.rb]: Didn't think of that. Might be interesting for the rest of the team as well @openSUSE/open-build-service

Comment (bgeuken) [inline src/api/test/functional/webui/package_controller_test.rb]: @hennevogel updated it now

File: src/api/spec/controllers/webui/request_controller_spec.rb [hunk 4]
```diff
@@ -76,4 +76,44 @@
...
+          devel_project: source_project.name, devel_package: source_package.name, description: "change it!"
+        @bs_request = BsRequest.where(description: "change it!", creator: submitter.login, state: "new").first
+      end
+
+      it { expect(response).to redirect_to(request_show_path(number: @bs_request)) }
+      it { expect(flash[:success]).to be nil }
+
+      it "creates a delete request with the correct data" do
+        expect(@bs_request).not_to be nil
+        expect(@bs_request.description).to eq("change it!")
```
Comment (ChrisBr) [inline src/api/spec/controllers/webui/request_controller_spec.rb]: Why not splitting this up in single ``expects``?
And also I'm wondering if a simple

``
expect { post }.to change{ BsRequest.count }.by(1)
``

wouldn't be sufficient?

File: src/api/spec/controllers/webui/request_controller_spec.rb [hunk 5]
```diff
@@ -76,4 +76,44 @@
...
+        request_action = @bs_request.bs_request_actions.where(
+          type: "change_devel",
+          target_project: target_project.name,
+          target_package: target_package.name,
+          source_project: source_project.name,
+          source_package: source_package.name
+        )
+        expect(request_action).to exist
```
Comment (ChrisBr) [inline src/api/spec/controllers/webui/request_controller_spec.rb]: I also would wrap this in a separate expect

File: src/api/spec/controllers/webui/request_controller_spec.rb [hunk 6]
```diff
@@ -76,4 +76,44 @@
...
+      before do
+        login(submitter)
+        post :change_devel_request, project: target_project.name, package: target_package.name,
+          devel_project: source_project.name, devel_package: "non-existant", description: "change it!"
+        @bs_request = BsRequest.where(description: "change it!", creator: submitter.login, state: "new").first
+      end
+
+      it { expect(flash[:error]).to eq("No such package: #{source_project.name}/non-existant") }
+      it { expect(response).to redirect_to(package_show_path(project: target_project, package: target_package)) }
+      it { expect(@bs_request).to be nil }
```
Comment (ChrisBr) [inline src/api/spec/controllers/webui/request_controller_spec.rb]: Or to ``to not change (BsRequest.count)``

**Comment by ChrisBr** [comment]
Otherwise LGTM

File: src/api/spec/controllers/webui/request_controller_spec.rb (hunk 6 above)
Comment (bgeuken) [inline src/api/spec/controllers/webui/request_controller_spec.rb]: But then I would have to wrap this around the post request. I wanted to avoid that.

**Comment by bgeuken** [comment]
@ChrisBr \*updated\*

File: src/api/spec/controllers/webui/request_controller_spec.rb [hunk 7]
```diff
@@ -76,4 +73,44 @@
...
+          devel_project: source_project.name, devel_package: source_package.name, description: "change it!"
+        @bs_request = BsRequest.where(description: "change it!", creator: submitter.login, state: "new").first
+      end
+
+      it { expect(response).to redirect_to(request_show_path(number: @bs_request)) }
+      it { expect(flash[:success]).to be nil }
+      it { expect(@bs_request).not_to be nil }
+      it { expect(@bs_request.description).to eq("change it!") }
+
+      it "creates a request with correct data" do
```
Comment (ChrisBr) [inline src/api/spec/controllers/webui/request_controller_spec.rb]: request action?

File: src/api/spec/controllers/webui/request_controller_spec.rb (hunk 6 above)
Comment (ChrisBr) [inline src/api/spec/controllers/webui/request_controller_spec.rb]: Yeah, right!
Ok, I'm fine with that

**Comment by bgeuken** [comment]
Christian gave me green light -> merged

**Pull request merged**